lib/projectStructure.ts

```diff
@@ -1,36 +1,103 @@
-// @ts-nocheck
-const fs = require('fs');
-const path = require('path');
-const { walk } = require('@hubspot/local-dev-lib/fs');
-const { logger } = require('@hubspot/local-dev-lib/logger');
-const { logError } = require('./errorHandlers/index');
-
-const COMPONENT_TYPES = Object.freeze({
+import * as fs from 'fs';
+import * as path from 'path';
+import { walk } from '@hubspot/local-dev-lib/fs';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { logError } from './errorHandlers/index';
+
+type ComponentTypes = 'private-app' | 'public-app' | 'hubl-theme';
+type ValueOf<T> = T[keyof T];
+
+export type Component = {
+  type: ComponentTypes;
+  config: object;
+  runnable: boolean;
+  path: string;
+};
+
+type PrivateAppComponentConfigType = {
+  name: string;
+  description: string;
+  uid: string;
+  scopes: Array<string>;
+  public: boolean;
+  extensions?: {
+    crm: {
+      cards: Array<{ file: string }>;
+    };
+  };
+};
+
+type PublicAppComponentConfigType = {
+  name: string;
+  uid: string;
+  description: string;
+  allowedUrls: Array<string>;
+  auth: {
+    redirectUrls: Array<string>;
+    requiredScopes: Array<string>;
+    optionalScopes: Array<string>;
+    conditionallyRequiredScopes: Array<string>;
+  };
+  support: {
+    supportEmail: string;
+    documentationUrl: string;
+    supportUrl: string;
+    supportPhone: string;
+  };
+  extensions?: {
+    crm: {
+      cards: Array<{ file: string }>;
+    };
+  };
+  webhooks?: {
+    file: string;
+  };
+};
+
+type AppCardComponentConfigType = {
+  type: 'crm-card';
+  data: {
+    title: string;
+    uid: string;
+    location: string;
+    module: {
+      file: string;
+    };
+    objectTypes: Array<{ name: string }>;
+  };
+};
+
+export const COMPONENT_TYPES = {
   privateApp: 'private-app',
   publicApp: 'public-app',
   hublTheme: 'hubl-theme',
-});
+} as const;
 
-const CONFIG_FILES = {
+export const CONFIG_FILES: {
+  [k in ValueOf<typeof COMPONENT_TYPES>]: string;
+} = {
   [COMPONENT_TYPES.privateApp]: 'app.json',
   [COMPONENT_TYPES.publicApp]: 'public-app.json',
   [COMPONENT_TYPES.hublTheme]: 'theme.json',
 };
 
-function getTypeFromConfigFile(configFile) {
-  for (const key in CONFIG_FILES) {
+function getTypeFromConfigFile(
+  configFile: ValueOf<typeof CONFIG_FILES>
```

Comment:
It looks like this just resolves to `string`. Which... I think is probably fine but we can just type at that way directly. I messed around with adding an `as const` to  `CONFIG_FILES` to fix this and it caused its own set of problems